images/recycler/scripts/recycler.sh

```diff
@@ -22,8 +21,10 @@ if [[ ! -d ${dir} ]]; then
     exit 1
 fi
 
-# shred all files
-find ${dir} -type f -exec shred -fuvz {} \;
+find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0
```

Comment:
you are right. I don't think `find` is able to go through directories with mode 0700. So we have to `sudo` to run `find` too.